extensions.html

```diff
@@ -445,16 +507,122 @@ <h2>
+          When the system receives new touch input values,
+          <dfn data-lt="updating touchEvents">update touchEvents</dfn>
+          by running the following steps:
+        </p>
+        <ol>
+          <li>Let |surfaceDimensionX:unsigned long| be an {{unsigned long}} value
+            representing the maximum x dimention of the touch surface on the |gamepad|.
+          </li>
+          <li>Let |surfaceDimensionY:unsigned long| be an {{unsigned long}} value
+            representing the maximum y dimention of the touch surface on the |gamepad|.
+          </li>
+          <li>Let |touchInputs:list| be a [=list=] of objects representing
+            the most recent logical touch input values reported by the |gamepad|.
+          </li>
+          <li>[=list/For each=] |touchData| in |touchInputs:list|:
+            <ol>
+              <li>
+                <p>Let |normalizedX:float| be a value calculated from the current
+                  touch input x coordinate in relationship to the surfaceDimensionX
+                  and normalized to the range [-1.0, 1.0] where -1.0 is the leftmost
+                  coordinate and 1.0 is the rightmost coordinate.</p>
+                <p>
+                <p>Possible implementation:</p>
+                <ul>
+                  <li>`normalizedX = (2.0 * touchData.x / surfaceDimensionX) - 1`</li>
+                </ul>
+                </p>
+              </li>
+              <li>
+                <p>Let |normalizedY:float| be a value calculated from the current
+                  touch input y coordinate in relationship to the surfaceDimensionY
+                  and normalized to the range [-1.0, 1.0] where -1.0 is the topmost
+                  coordinate and 1.0 is the bottommost coordinate.
+                </p>
+                <p>
+                <p>Possible implementation:</p>
+                <ul>
+                  <li>`normalizedY = (2.0 * touchData.y / surfaceDimensionY) - 1`</li>
+                </ul>
+                </p>
+              </li>
+              <li>Let |touchEvent:GamepadTouch| be a {{GamepadTouch}}.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/touchId}} to be
+              `touchData.touchID`.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/surfaceId}} to be
+              device surfaceId.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/position}}[0] to be
+              |normalizedX:float|.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/position}}[1] to be
+              |normalizedY:float|.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/surfaceDimensions}}[0] to be
+              device max x-axis coordinate.
+              </li>
+              <li>Set |touchEvent|.{{GamepadTouch/surfaceDimensions}}[1] to be
+              device max y-axis coordinate.<p>(<i>note: surfaceDimensions can be set to `null`</i>)</p>
```

Comment:
Updated.

Reply:
This may need more discussion. Perhaps there is not a case where surfaceDimensions can be null. For the DS4 touch coordinates are normalized using the device touch point and the surface dimensions. Should the algorithm include receiving input from a device that sends already normalized touch points and does not report surface dimensions?